### DISCUSSION TITLE
Will segments in the same shard be distributed to different query nodes?

### DISCUSSION THREAD

**Discussion by XiaominZou** [milvus-io/milvus#29763] [Q&A and General discussion]
Will segments of the same shard be distributed to different query nodes? If so, there will lead to an extra network round-trip time than 
 the way that keeps all segments of the shard in one querynode

**Comment by xiaofan-luan** [comment]
In short, yes.

Once segments becomes sealed it will be balanced in the resource group

**Reply by XiaominZou** [reply]
What does the resource group contains？If milvus only has one replica，will segments of the same shard be distributed to different query nodes?

**Reply by yhmo** [reply]
A resource group is a group of query nodes, you can specify how many query nodes are assigned to a resource group, and you can specify how many replicas are loaded to a resource group.

Sealed segments of the same shard can be distributed to different query nodes. Growing segments of the same shard will be kept in the same query node(the shard-leader node).

**Reply by XiaominZou** [reply]
Thanks!
